sdk/eventhub/event-hubs/src/connectionContext.ts

```diff
@@ -169,39 +198,52 @@ export namespace ConnectionContext {
       connectionContext.connection.removeAllSessions();
 
       // Close the cbs session to ensure all the event handlers are released.
-      await connectionContext.cbsSession.close();
+      await connectionContext.cbsSession.close().catch((err) => {
+        logger.verbose(
+          "[%s] Error when closing cbsSession after disconnected event: %O",
+          connectionContext.connection.id,
+          err
+        );
+      });
       // Close the management session to ensure all the event handlers are released.
-      await connectionContext.managementSession!.close();
+      await connectionContext.managementSession!.close().catch((err) => {
+        logger.verbose(
+          "[%s] Error when closing managementSession after disconnected event: %O",
+          connectionContext.connection.id,
+          err
+        );
+      });
 
       // Close all senders and receivers to ensure clean up of timers & other resources.
       if (state.numSenders || state.numReceivers) {
         for (const senderName of Object.keys(connectionContext.senders)) {
           const sender = connectionContext.senders[senderName];
-          if (!sender.isConnecting) {
```

Comment:
Why is this if check removed?

Reply:
It's possible for `disconnected` to be emitted while connecting. For example, if the connection is  viewed as being idle while the connection is still in the process of opening. This check was preventing us from closing the existing senders in this scenario.

Reply:
I think this check is a remnant of the time when we recovered sender links when connection was disconnected. So the code below where we have `sender.close()` was previously `sender.onDetached()`. Since that need to be done if the sender is already connecting, we had this check. It should have been removed when we made the decision to not re-connect broken sender links in the background i.e. in https://github.com/Azure/azure-sdk-for-js/pull/6737